src/Sulu/Component/Content/SmartContent/PageDataProvider.php

```diff
@@ -167,7 +167,19 @@ public function resolveDatasource($datasource, array $propertyParameter, array $
             return;
         }
 
-        return new DatasourceItem($result[0]['id'], $result[0]['title'], '/' . ltrim($result[0]['path'], '/'));
+        $path = null;
+        if ($this->showDrafts) {
+            // only show the path in the sulu admin ui when
+            if (isset($result[0]['path'])) {
+                $path = '/' . ltrim($result[0]['path'], '/');
+            } elseif (isset($result[0]['nodePath'])) {
+                $pathParts = explode('/', $result[0]['nodePath']);
+                $pathParts = array_slice($pathParts, 4);
+                $path = '/' . implode('/', $pathParts);
+            }
+        }
+
+        return new DatasourceItem($result[0]['id'], $result[0]['title'], $path);
```

Comment:
Didn't we make the mistake we were trying to avoid here on our own? Shouldn't we be passing the URL after all? :thinking: If you think about it, the `path` is nothing the user can handle, especially when he is not managing the UI in a different language than the default language...